node/src/database.rs

```diff
@@ -207,3 +214,24 @@ impl Default for DatabaseOptions {
         }
     }
 }
+
+/// The Archivist is responsible for storing the events and potentially other
+/// ephemeral data forever in the archive DB.
+///
+/// Example:
+/// - The block does not store the events, only the hash of the events. The
+///   archivist will store the events.
+#[cfg(feature = "archive")]
+pub(crate) trait Archivist {
```

Comment:
if its a trait that's just being used inside the crate and not exported, don't you think it will be more readable with less abstraction and more functional. This is just design philosophy but traits are understandable if they are exported (we're also trying to reduce the number of traits in the wallets from Store/State